karapace/backup/consumer.py

```diff
@@ -0,0 +1,45 @@
+"""
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from datetime import timedelta
+from isodate import duration_isoformat, parse_duration
+from typing import Union
+
+__all__ = ["PollTimeout"]
+
+
+# TODO @final (Python 3.8+)
+class PollTimeout:
+    """Specifies how long a single poll attempt may take while consuming the topic.
+
+    It may be necessary to adjust this value in case the cluster is slow. The value must be given in ISO8601 duration
+    format (e.g. `PT1.5S` for 1,500 milliseconds) and must be at least on second. Defaults to one minute.
+    """
+
+    __slots__ = ("__value",)
+
+    def __init__(self, value: Union[str, timedelta]) -> None:
+        self.__value = value if isinstance(value, timedelta) else parse_duration(value)
+        if self.__value // timedelta(seconds=1) < 1:
+            raise ValueError(f"Poll timeout MUST be at least one second, got: {self}")
+
+    @classmethod
+    def default(cls) -> "PollTimeout":
```

Comment:
nit (non-blocking): I think we should prefer `from __future__ import annotations` over string syntax type hints, so that we're more compatible with the future than with the past.